### ISSUE TITLE
[Feature] Integrate Differential Correction in Wavelet domain (DCW) sampler hook to mitigate SNR-t bias

### ISSUE THREAD

**Issue by ChuxiJ** [ace-step/ACE-Step-1.5#1119]
## Summary

This issue proposes integrating **DCW (Differential Correction in Wavelet domain)** — a CVPR 2026 sampler-side technique — into ACE-Step-1.5's flow-matching DiT loop as an opt-in quality knob. DCW is training-free, adds negligible compute, and has been validated on a wide range of diffusion models **including FLUX**, which shares ACE-Step's flow-matching formulation.

- Paper: *Elucidating the SNR-t Bias of Diffusion Probabilistic Models* — arXiv:2604.16044
- Reference code: https://github.com/AMAP-ML/DCW
- Citation: Yu, Sun, Zeng, Chu, Zhan — CVPR 2026

## Background: what is SNR-t bias?

During **training**, the SNR of a noised sample `x_t` is strictly coupled to its timestep `t`. During **inference**, because we step through a predicted trajectory rather than the true forward process, the actual SNR of `x_t` drifts away from what the network was trained to see at that `t`. The mismatch accumulates along the denoising chain and degrades output quality — most visibly at intermediate / high timesteps.

The authors' key observation: diffusion models reconstruct **low-frequency content before high-frequency content** during the reverse process. Therefore the SNR-t drift affects different frequency bands differently, and the correction should be applied **per-band** (wavelet domain) rather than in pixel / latent space directly.

## The DCW method in one screen

Per sampler step, given current latent `x`, predicted velocity `v`, timestep `t`:

```
x_next   = sampler_step(x, v, t)        # normal Euler / Heun / flow-matching update
denoised = x - v * t                    # predicted clean sample x_0

# decompose both with a 1-level DWT
xL, xH   = DWT(x_next)
yL, yH   = DWT(denoised)

# Low-frequency mode (Eq. 18 / 20):
xL       = xL + scaler * (xL - yL)
x_next   = IDWT(xL, xH)

# High-frequency and "double" modes follow the same pattern on xH / both bands.
```

For flow-matching models (the paper's FLUX recipe, which is closest to ACE-Step), the scaler is typically decayed with `t`: `s = (t_curr / sigma_max) * corr_scaler`. Any wavelet basis works — the authors evaluate `haar`, `db4`, `sym8`.

## Why this fits ACE-Step-1.5

ACE-Step-1.5 uses a **custom flow-matching** Euler/Heun/SDE sampler over 1-D audio latents of shape `[B, T, 64]` at 25 Hz. The per-step update lives in:

- PyTorch path: `acestep/models/base/modeling_acestep_v15_base.py::generate_audio` — main loop at lines 1989–2131
- MLX path: `acestep/models/mlx/dit_generate.py` — main loop at lines 343–405

Because the latents are 1-D temporal, we apply a **1-D DWT** along the `T` axis (transpose `[B, T, C] → [B, C, T]`, run `DWT1DForward`, apply correction, `DWT1DInverse`, transpose back). Everything else maps 1:1 from the FLUX recipe.

DCW sits naturally alongside the existing sampler stabilisation knobs (`velocity_norm_threshold`, `velocity_ema_factor`) that were added in #957.

## Proposed API

New optional fields on `GenerationParams` in `acestep/inference.py`, forwarded through the handler chain into `generate_audio`:

| Field | Type | Default | Description |
|---|---|---|---|
| `dcw_enabled` | `bool` | `False` | Master switch (off by default, preserves current behaviour bit-for-bit). |
| `dcw_mode` | `str` | `"low"` | One of `"low"`, `"high"`, `"double"`, `"pix"`. |
| `dcw_scaler` | `float` | `0.1` | Low-band correction strength (or the single scaler for `"high"` / `"pix"`). |
| `dcw_high_scaler` | `float` | `0.0` | High-band correction strength (used when `dcw_mode == "double"`). |
| `dcw_wavelet` | `str` | `"haar"` | PyWavelets name — e.g. `"haar"`, `"db4"`, `"sym8"`. |

The scaler is internally modulated by `t_curr` to match the FLUX-DCW recipe (`effective_s = t_curr * dcw_scaler`), so a constant user-facing value behaves reasonably across the whole denoising trajectory.

## Dependency

DCW needs `pytorch_wavelets` (PyTorch frontend) and `PyWavelets` (the underlying wavelet kernels). These are **not** added as hard deps — the module imports them lazily and falls back with a clear warning if they are missing and DCW is enabled. The README will recommend `pip install pytorch_wavelets PyWavelets` for users who want to try the feature.

## Scope of the initial PR

To keep the first PR reviewable:

- ✅ New shared module `acestep/models/common/dcw_correction.py` with `dcw_low / dcw_high / dcw_double / dcw_pix` helpers and a `DCWCorrector` class that caches DWT modules per device/dtype.
- ✅ Unit tests for the module (roundtrip, zero-scaler identity, low/high band isolation).
- ✅ Hook into the **base** model's `generate_audio` PyTorch loop — covers Euler, Heun and SDE modes.
- ✅ Forward the new params through `GenerationParams` → handler chain → `generate_audio`.
- ✅ Short docs page with usage example and parameter guide.

## Follow-up work (tracked here; separate PRs)

- [ ] Propagate the DCW hook to the **turbo**, **sft**, **xl_base**, **xl_sft**, **xl_turbo** variants of `modeling_acestep_v15_*.py`.
- [ ] Port the same hook into the **MLX** path (`acestep/models/mlx/dit_generate.py`). MLX doesn't ship wavelets, so we'll need an MLX-native Haar implementation or a short `mx.array ↔ torch` bridge only when DCW is enabled.
- [ ] Expose the new params in the **Gradio UI** under the existing "Advanced DiT controls" panel (`acestep/ui/gradio/interfaces/generation_advanced_dit_controls.py`) and translate the labels in `acestep/ui/gradio/i18n/*.json`.
- [ ] Benchmark on the standard prompts in `examples/` to report objective quality deltas at `inference_steps=8, 16, 32`.

## Acceptance criteria

1. `dcw_enabled=False` produces bit-identical output to main (no-op when disabled).
2. With `pytorch_wavelets` installed and `dcw_enabled=True, dcw_mode="low", dcw_scaler=0.1`, a standard 32-step generation runs end-to-end without errors and the per-step overhead is < 2 %.
3. If `pytorch_wavelets` is missing and DCW is enabled, the pipeline logs a clear warning and falls back to the normal sampler (no crash).

I'll open a PR shortly that implements the initial scope above.

**Comment by ServeurpersoCom** [comment]
<img width="432" height="446" alt="Image" src="https://github.com/user-attachments/assets/991d327d-61ca-40fd-a85d-b546aae649c9" />
We'll test it together :D

**Comment by ServeurpersoCom** [comment]
Stock XL Turbo
[Rock de la gerbe (xl-turbo text2music) - STOCK.mp3](https://github.com/user-attachments/files/26939732/Rock.de.la.gerbe.xl-turbo.text2music.-.STOCK.mp3)

XL Turbo - DCW mode = Low - DCW scaler = 0.05 - Very small morphing, better medium/treble spectrum !
[Rock de la gerbe (xl-turbo text2music) - DCW mode = Low - DCW scaler  = 0.05.mp3](https://github.com/user-attachments/files/26939736/Rock.de.la.gerbe.xl-turbo.text2music.-.DCW.mode.Low.-.DCW.scaler.0.05.mp3)

(The lyrics of the song are dumb humorous in French)
